homeassistant/components/temper/sensor.py

```diff
@@ -102,3 +106,30 @@ def update(self) -> None:
                 "have changed. Attempting to reset device"
             )
             reset_devices()
+
+class HumiditySensor(SensorEntity):
+    """Representation of a Temper humidity sensor."""
+
+    _attr_device_class = SensorDeviceClass.HUMIDITY
+    _attr_native_unit_of_measurement = PERCENTAGE
+
+    def __init__(self, temper_device, name):
+        """Initialize the sensor."""
+        self.set_temper_device(temper_device)
+
+        self._attr_name = f"{name} humidity"
+        self._attr_unique_id = f"temper_{temper_device.type.name}_humidity"
+
+    def set_temper_device(self, temper_device):
+        """Assign the underlying device for this sensor."""
+        self.temper_device = temper_device
+
+    def update(self) -> None:
+        """Retrieve latest state."""
+        try:
+            sensor_value = self.temper_device.get_humidity()[0]['humidity_pc']
+            self._attr_native_value = round(sensor_value, 1)
```

Comment:
Rounding shouldn't be done in the integration anymore, instead, set the suggested_precision.

Reply:
:+1: Good idea, I updated this to use `SensorEntityDescription` with a `suggested_display_precision`